tests/post/steps/test_service_configuration.py

```diff
@@ -0,0 +1,271 @@
+import yaml
+
+import pytest
+from pytest_bdd import scenario, given, then, when, parsers
+
+import kubernetes.client
+from kubernetes.client import AppsV1Api
+from kubernetes.client.rest import ApiException
+
+from tests import utils
+
+
+# Fixtures {{{
+
+
+@pytest.fixture
+def k8s_appsv1_client(k8s_apiclient):
+    return AppsV1Api(api_client=k8s_apiclient)
+
+@pytest.fixture
+def teardown(host, version, request, k8s_client):
+    yield
+    teardown_configmap(k8s_client)
+
+
+# }}}
+
+
+# Scenarios {{{
+
+
+@scenario('../features/service_configuration.feature',
+          'Propagation of Service Configurations to underlying Services')
+def test_service_config_propagation(host, teardown):
+    pass
+
+
+# }}}
+
+
+# Given {{{
+
+
+@given(parsers.parse(
+    "we have a '{name}' CSC in namespace '{namespace}' with "
+    "'{path}' equal to '{value}'"))
+def check_csc_configuration(
+    host,
+    k8s_client,
+    name,
+    namespace,
+    path,
+    value
+):
+    csc_response = read_namespace_config_map(k8s_client, name, namespace)
+
+    assert csc_response, (
+        "No ConfigMap with name {} in namespace {} found".format(
+            name, namespace
+        )
+    )
+
+    csc_obj = yaml_load(csc_response, name, namespace)
+    response_value = traversed_dict(csc_obj, path)
+
+    assert int(value) == response_value, (
+        "Expected value {} for key {} in ConfigMap {} found in namespace {}, "
+        "got {}".format(value, path, name, namespace, response_value)
+    )
+
+
+# }}}
+
+
+# When {{{
+
+
+@when(parsers.parse(
+    "we update '{name}' CSC in namespace '{namespace}' "
+    "'{path}' to '{value}'"))
+def update_service_configuration(
+    host,
+    k8s_client,
+    name,
+    namespace,
+    path,
+    value
+):
+
+    full_csc = read_namespace_config_map(k8s_client, name, namespace)
+    csc_obj = yaml_load(full_csc, name, namespace)
+
+    csc_obj['spec']['deployment']['replicas'] = int(value)
+    patch = \
+        {'data': {'config.yaml': yaml.safe_dump(
+            csc_obj, default_flow_style=False)
+        }}
```

Comment:
Minor: Personnaly don't really like this indentation
```suggestion
    patch = {
        'data': {
            'config.yaml': yaml.safe_dump(csc_obj, default_flow_style=False)
        }
    }
```